griptape/tools/google_docs/tool.py

```diff
@@ -0,0 +1,302 @@
+                    ): str,
+                    Literal(
+                        "text",
+                        description="Text to be appended to the Google Doc."
+                    ): str
+                }
+            ),
+        }
+    )
+    def append_text_to_google_doc(self, params: dict) -> InfoArtifact | ErrorArtifact:
+        try:
+            file_path = params["file_path"]
+            text = params["text"]
+            service = self._build_client(self.DOCS_SCOPES, "docs", "v1")
+            drive_service = self._build_client(self.DRIVE_SCOPES, "drive", "v3")
+            document_id = self._path_to_file_id(drive_service, file_path)
+            if document_id:
+                doc = service.documents().get(documentId=document_id).execute()
+                content = doc["body"]["content"]
+                last_text = content[-1]["paragraph"]["elements"][-1]["textRun"]["content"]
+                append_index = content[-1]["endIndex"]
+                if last_text.endswith("\n"):
+                    append_index -= 1
+
+                requests = [{"insertText": {"location": {"index": append_index}, "text": text}}]
+
+                service.documents().batchUpdate(documentId=document_id, body={"requests": requests}).execute()
+                return InfoArtifact("text appended successfully")
+            else:
+                return ErrorArtifact(f"error appending to Google Doc, file not found for path {file_path}")
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"error appending text to Google Doc with path {file_path}: {e}")
+
+    @activity(
+        config={
+            "description": "Can be used to prepend text to a Google Doc",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_path",
+                        description="Destination file path in the POSIX format. For example, 'foo/bar/baz.txt'"
+                    ): str,
+                    Literal(
+                        "text",
+                        description="Text to be prepended to the Google Doc."
+                    ): str
+                }
+            ),
+        }
+    )
+    def prepend_text_to_google_doc(self, params: dict) -> InfoArtifact | ErrorArtifact:
+        try:
+            file_path = params["file_path"]
+            text = params["text"]
+            service = self._build_client(self.DOCS_SCOPES, "docs", "v1")
+            drive_service = self._build_client(self.DRIVE_SCOPES, "drive", "v3")
+            document_id = self._path_to_file_id(drive_service, file_path)
+            if document_id:
+                doc = service.documents().get(documentId=document_id).execute()
+
+                if len(doc["body"]["content"]) == 1:
+                    requests = [{"insertText": {"location": {"index": 1}, "text": text}}]
+                else:
+                    start_index = doc["body"]["content"][1]["startIndex"]
```

Comment:
What happens if `len(doc["body"]["content"])` is 0?

Reply:
It still appends/prepends